mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisStorageProvider.java

```diff
@@ -27,6 +27,7 @@
 import io.mantisrx.server.master.resourcecluster.TaskExecutorID;
 import io.mantisrx.server.master.resourcecluster.TaskExecutorRegistration;
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 import rx.Observable;
```

Comment:
I think this is a very nice clean up. I suggest adding details javadoc comment here and in MantisStorageProvider explaining the differences between these 2 interfaces. For me, that has been a source of confusion.

Reply:
+1

Reply:
Can you add java doc?

Reply:
I suggestion I have would be to rename classes in this package XXXPersistenceProvider, and classes in the other package XXXStoreProvider. Drop the use of 'Storage'. Also This is a good opportunity to have consistent naming. So for interfaces, you can rename to `IKeyValueStoreProvider`.